lib/clouds/shared_functions.py

```diff
@@ -1247,38 +1079,115 @@ def populate_cloudconfig(self, obj_attr_list) :
                         cloudconfig += "      up /etc/openvpn/client_connected.sh\n"
                 fh.close()
 
-        cloudconfig += """
-runcmd:
-  - chmod +x /tmp/userscript.sh"""
-
-        # We can't run the userdata from cloudbench until the VPN is connected,
-        # so only run it if we're not using the VPN.
-        # Otherwise, /etc/openvpn/client_connected.sh will do it.
-        if obj_attr_list["use_vpn_ip"].lower() == "false" :
-            cloudconfig += """
-  - /tmp/userscript.sh"""
-        else :
-            cloudconfig += """
-  - chmod +x /etc/openvpn/client_connected.sh
-  - mv """ + conf_destination + """ /tmp/cbvpn.conf
-  - rm -f /etc/openvpn/*.conf /etc/openvpn/*.ovpn
-  - mv /tmp/cbvpn.conf """ + conf_destination + """
-  - mkdir -p /var/log/openvpn
-  - openvpn --daemon --config """ + conf_destination + """
-"""
-  #- systemctl start openvpn@""" + obj_attr_list["cloud_name"] + """_client-cb-openvpn-cloud.service
-  #- service openvpn start
+        if obj_attr_list["userdata_post_boot"].lower() == "true" or obj_attr_list["use_vpn_ip"].lower() != "false" :
```

Comment:
Same thing here. Using explicit newlines is prone to error for both of us, and it becomes increasingly difficult to remember when to use a newline and when not to use a newline.

Multi-line strings have all the same abilities as regular strings.

Reply:
Same comment as above. Will bring it back if you insist, no problem at all.